java/fury-core/src/main/java/org/apache/fury/serializer/AbstractObjectSerializer.java

```diff
@@ -256,6 +306,29 @@ private InternalFieldInfo[] buildFieldsInfo() {
     return fieldInfos;
   }
 
+  public static InternalFieldInfo[] buildFieldsInfo(Fury fury, List<Field> fields) {
```

Comment:
Could we just use `org.apache.fury.serializer.AbstractObjectSerializer#buildFieldInfos` instead. We should only copy those fields if we create object new instance following serialization process